tests/test_AggregateConnectionData.cpp

```diff
@@ -792,6 +795,43 @@ BOOST_AUTO_TEST_CASE(InactiveCell) {
     }
 }
 
+BOOST_AUTO_TEST_CASE(TestRestartIOConnection) {
+    std::vector<bool> lh(1000);
+    std::vector<double> dh(1000);
+    auto simCase = SimulationCase{first_sim()};
+    const auto rptStep = std::size_t{1};
+    const auto ih = MockIH {static_cast<int>(simCase.sched.getWells(rptStep).size())};
+    const Opm::data::WellRates wrc = wr();
+
+    Opm::RestartIO::Header header(ih.value, lh, dh);
+    auto conn = Opm::RestartIO::Helpers::AggregateConnectionData{ih.value};
+    conn.captureDeclaredConnData(simCase.sched,
+                                 simCase.grid,
+                                 simCase.es.getUnits(),
+                                 wrc,
+                                 rptStep
+                                 );
+
+    const auto icon = conn.getIConn();
+    const auto scon = conn.getSConn();
+    const auto xcon = conn.getXConn();
+
+    std::vector<Opm::RestartIO::Connection> connections;
+    for (std::size_t iw=0; iw < header.num_wells; iw++) {
+        for (std::size_t ic=0; ic < header.ncwmax; ic++) {
```

Comment:
This is test code so probably not a big deal.  In general, though, the well's number of active connections is IWEL[[`IWell::NConn`](https://github.com/OPM/opm-common/blob/75b64025a321072a40b75b86dc30937cc59539e9/opm/output/eclipse/VectorItems/well.hpp#L33)]

Reply:
Yes I know; the test is written "before" IWEL is created.